src/Target.cpp

```diff
@@ -132,8 +132,85 @@ string get_env(const char *name) {
     }
 #endif
 }
+
+const std::map<std::string, Target::OS> os_name_map = {
+    {"os_unknown", Target::OSUnknown},
+    {"linux", Target::Linux},
+    {"windows", Target::Windows},
+    {"osx", Target::OSX},
+    {"android", Target::Android},
+    {"ios", Target::IOS},
+    {"nacl", Target::NaCl},
+};
+
+bool lookup_os(const std::string &tok, Target::OS &result) {
+    auto os_iter = os_name_map.find(tok);
+    if (os_iter != os_name_map.end()) {
+        result = os_iter->second;
+        return true;
+    }
+    return false;
+}
+
+const std::map<std::string, Target::Arch> arch_name_map = {
+    {"arch_unknown", Target::ArchUnknown},
+    {"x86", Target::X86},
+    {"arm", Target::ARM},
+    {"pnacl", Target::PNaCl},
+    {"mips", Target::MIPS},
+};
+
+bool lookup_arch(const std::string &tok, Target::Arch &result) {
+    auto arch_iter = arch_name_map.find(tok);
+    if (arch_iter != arch_name_map.end()) {
+        result = arch_iter->second;
+        return true;
+    }
+    return false;
+}
+
+const std::map<std::string, Target::Feature> feature_name_map = {
```

Comment:
The map guarantees the tokens are in alphabetical order within their category (arch, os, feature), where before they were in enum order. I feel that for human readability, alphabetical order is preferable. I had to change the test to deal with this change, though I would be surprised if anything else depends on the ordering and I'm a bit annoyed that the test does, but fixing that either requires relying on Target itself to reparse the string or implementing an tokenizer in the test, both of which seem counterproductive.